docs/pages/router/migrate/from-react-navigation.mdx

```diff
@@ -0,0 +1,468 @@
+-    route
+}) {
+-    const user = route?.params?.user;
+
++    const { user } = useLocalSearchParams();
+}
+```
+
+### Migrating the Link component
+
+React Navigation and Expo Router both provide Link components. However, Expo's Link component uses `href` instead of [`to`](https://reactnavigation.org/docs/use-link-props#to).
+
+```jsx
+// React Navigation
+<Link to="Settings" />
+
+// Expo Router
+<Link href="/settings" />
+```
+
+React Navigation users will often create a custom Link component with the `useLinkProps` hook in order to control the child component. This isn't neccessary in Expo Router, instead use the `asChild` prop.
+
+### Sharing screens across navigators
+
+It's common for React Navigation apps to reuse a set of routes across multiple navigators. This is generally used with tabs to ensure each tab can push any screen.
+
+In Expo Router, you can either migrate to [shared routes](/router/advanced/shared-routes) or create multiple files and re-export the same component from them.
+
+When you use groups or shared routes, you can navigate to specific tabs by using the fully qualified route name, for example, `/(home)/settings` instead of `/settings`.
+
+### Migrating screen tracking events
+
+You may have your screen tracking setup according to our [React Navigation screen tracking guide](https://reactnavigation.org/docs/screen-tracking/), update it according to the [Expo Router screen tracking guide](/router/reference/screen-tracking).
+
+### Using platform-specifc components for screens
+
+Refer to the [platform-specific modules](/router/advanced/platform-specific-modules) guide for info on switching UI based on the platform.
+
+### Replacing the NavigationContainer
+
+The global React Navigation [`<NavigationContainer />`](https://reactnavigation.org/docs/navigation-container/) is completely managed in Expo Router. Expo Router provides systems for achieving the same functionality as the `NavigationContainer` without needing to use it directly.
+
+<Collapsible summary="API substitutions">
+
+### ref
+
+The `NavigationContainer` ref should not be accessed directly. Use the following methods instead.
+
+#### `resetRoot​`
+
+Navigate to the initial route of the application. For example, if your app starts at `/` (recommended), then you can replace the current route with `/` using this method.
+
+```js
+import { useRouter } from 'expo-router';
+
+function Example() {
+  const router = useRouter();
+
+  return (
+    <Text
+      onPress={() => {
+        // Go to the initial route of the application.
+        router.replace('/');
+      }}>
+      Reset App
+    </Text>
+  );
+}
+```
+
+#### `getRootState`
+
+Use `useRootNavigationState()`.
+
+#### `getCurrentRoute`
+
+Unlike React Navigation, Expo Router can reliably represent any route with a string. Use the [`usePathname()`](/router/reference/screen-tracking/) or [`useSegments()`](/router/reference/typed-routes/#relative-paths) hooks to identify the current route.
```

Comment:
```suggestion
Unlike React Navigation, Expo Router can reliably represent any route with a string. Use the [`usePathname()`](/router/reference/hooks/#usepathname) or [`useSegments()`](/router/reference/hooks/#usesegments) hooks to identify the current route.
```

Updating internal links from https://github.com/expo/expo/pull/23355/files so that we can merge them together or unblock this PR.